src/components/calculator/selectors/InteractionTypeSelector.tsx

```diff
@@ -0,0 +1,160 @@
+  const getInteractionRiskMessage = (interactionType: string) => {
+    if (interactionType === 'oneTime' || interactionType === 'workplace') {
+      return 'calculator.one_time_interaction_risk_message'
+    } else if (interactionType === 'repeated') {
+      return 'calculator.repeated_interaction_risk_message'
+    } else if (interactionType === 'partner') {
+      return 'calculator.partner_interaction_risk_message'
+    }
+    console.warn(
+      'No interaction risk message for interaction type ',
+      interactionType,
+    )
+    return 'calculator.one_time_interaction_risk_message'
+  }
+
+  const formattedInteractionTypeName = props.showResetButton ? (
+    <Dropdown>
+      <Dropdown.Toggle
+        id="custom-interaction-type"
+        variant="light"
+        className="text-wrap"
+      >
+        {t(Interaction[props.interactionType].label)}
+      </Dropdown.Toggle>
+      <Dropdown.Menu className="w-100">
+        {Object.keys(Interaction).map((value, index) => {
+          return (
+            <Dropdown.Item
+              key={index}
+              className="text-wrap"
+              onSelect={() => {
+                props.setInteractionType(value)
+              }}
+            >
+              {Interaction[value].label}
+            </Dropdown.Item>
+          )
+        })}
+      </Dropdown.Menu>
+    </Dropdown>
+  ) : (
+    <span>
+      {props.interactionType === undefined || props.interactionType === ''
+        ? t('calculator.set_interaction_type_to_continue')
+        : t(Interaction[props.interactionType].label)}
+    </span>
+  )
+
+  return (
+    <>
+      <div className="d-inline-block d-md-inline">
+        {formattedInteractionTypeName}
+      </div>
+      {props.interactionType !== undefined && !props.editInteractionType && (
+        <Button
+          onClick={() => setShowInteractionTypeMath(!showInteractionTypeMath)}
+          aria-controls="interaction-type-math"
+          aria-expanded={showInteractionTypeMath}
+          variant="link"
+          size="sm"
+          className="pl-0 font-weight-bold d-block underline-for-focus"
+        >
+          {showInteractionTypeMath ? (
+            <BsChevronDown className="align-baseline" />
+          ) : (
+            <BsChevronRight className="align-baseline" />
+          )}
+          {showInteractionTypeMath
+            ? t('calculator.hide_math')
+            : t('calculator.show_math')}
+        </Button>
+      )}
+      {props.interactionType !== undefined && !props.editInteractionType && (
+        <Collapse in={showInteractionTypeMath}>
+          <ul>
+            <li>
+              <Trans
+                i18nKey={getInteractionRiskMessage(props.interactionType)}
+                values={{
+                  percentage: fixedPointPrecisionPercent(
+                    Interaction[props.interactionType].multiplier,
+                  ),
+                }}
+              >
+                Blah <strong>blah</strong>
```

Comment:
Replace Blah with the gist of what will be rendered.

Reply:
Done